modules/applyRouterMiddleware.js

```diff
@@ -0,0 +1,34 @@
+import React, { createElement } from 'react'
+import RouterContext from './RouterContext'
+
+export default (...middlewares) => {
+  const withContextContainer = middlewares.filter(
```

Comment:
Slightly more optimized to write this as

```js
middlewares.map(m => m.renderRouteComponent).filter(f => f)
```

and then skip destructuring in the reduce callback below.